packages/shared/src/components/post/PostContentHeader.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { ReactElement } from 'react';
+import { useRouter } from 'next/router';
+import { Button, ButtonSize, ButtonVariant } from '../buttons/Button';
+import { ArrowIcon } from '../icons';
+import { PostHeaderActions } from './PostHeaderActions';
+import { PostNavigationProps } from './common';
+import { SourceType } from '../../graphql/sources';
+
+export function PostContentHeader({
+  post,
+  onReadArticle,
+}: Pick<PostNavigationProps, 'onReadArticle' | 'post'>): ReactElement {
+  const router = useRouter();
+  const canGoBack = !!globalThis?.window?.history?.length;
+  const routedFromSquad = router?.query?.squad;
+  const squadLink = `/squads/${router.query.squad}`;
+
+  if (!canGoBack && !routedFromSquad && post.source.type !== SourceType.Squad) {
```

Comment:
Should it not be `||` on the sourcetype?
I think the logic is something like even if you end from google on a squad post we still show back to squad there right?

Reply:
> Should it not be || on the sourcetype?

The issue would be the same for any type, the reason Squad is allowed is we can go back to its Squad itself.

> if you end from google on a squad post we still show back to squad there right?

Yeah, was thinking of that too, I tried to check whether the back history is within the app, but didn't find any quick way.

Reply:
> That's what we do at the moment, so I believe that should be the case here as well. Just not in this condition

It is still the same condition (PostBackButton) we have in the past, just mixed all together now rather than being on different components.